azure_pipelines/spm-framework.yml

```diff
@@ -0,0 +1,311 @@
+- job: BuildXcFrameworks
+  displayName: Build MSAL framework and release
+  pool:
+    vmImage: 'macOS-latest'
+    timeOutInMinutes: 20
+
+  steps:
+  - checkout: msalRepository
+    clean: true
+    submodules: true
+    fetchDepth: 1
+    persistCredentials: true
+  - task: Xcode@5
+    displayName: Build archive for iOS Simulator
+    inputs:
+      actions: 'archive'
+      sdk: 'iphonesimulator'
+      xcWorkspacePath: 'MSAL.xcworkspace'
+      scheme: 'MSAL (iOS Framework)'
+      packageApp: false
+      args: 'SKIP_INSTALL=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES -archivePath $(Build.binariesDirectory)/iOS-Sim'
+      exportPath: $(Build.binariesDirectory)/iOS-Sim
+  - task: Xcode@5
+    displayName: Build archive for iOS device
+    inputs:
+      actions: 'archive'
+      sdk: 'iphoneos'
+      xcWorkspacePath: 'MSAL.xcworkspace'
+      scheme: 'MSAL (iOS Framework)'
+      packageApp: false
+      destinationTypeOption: 'devices'
+      args: 'SKIP_INSTALL=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES -archivePath $(Build.binariesDirectory)/iOS'
+      exportPath: $(Build.binariesDirectory)/iOS
+  - task: Xcode@5
+    displayName: Build archive for macOS
+    inputs:
+      actions: 'archive'
+      sdk: 'macosx'
+      xcWorkspacePath: 'MSAL.xcworkspace'
+      scheme: 'MSAL (Mac Framework)'
+      packageApp: false
+      args: 'SKIP_INSTALL=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES -archivePath $(Build.binariesDirectory)/macOS'
+      exportPath: $(Build.binariesDirectory)/macOS
+  - task: Bash@3
+    displayName: Build xcframework from archives
+    inputs:
+      workingDirectory: $(Build.binariesDirectory)
+      targetType: 'inline'
+      script: |
+        # Building xcframework
+        cd $(Build.BinariesDirectory)
+        
+        echo "Build Dir = $(pwd)"
+
+        xcodebuild -create-xcframework \
+        -framework $(Build.binariesDirectory)/iOS.xcarchive/Products/Library/Frameworks/MSAL.framework \
+        -framework $(Build.binariesDirectory)/iOS-Sim.xcarchive/Products/Library/Frameworks/MSAL.framework \
+        -framework $(Build.binariesDirectory)/macOS.xcarchive/Products/Library/Frameworks/MSAL.framework \
+        -output $(Build.binariesDirectory)/MSAL.xcframework
+      failOnStderr: true
+  - task: ArchiveFiles@2
+    displayName: Zip xcframework
+    inputs:
+      rootFolderOrFile: '$(Build.binariesDirectory)/MSAL.xcframework'
+      includeRootFolder: true
+      archiveType: 'zip'
+      archiveFile: '$(Build.ArtifactStagingDirectory)/MSAL.zip'
+      replaceExistingArchive: true
+  - task: EsrpCodeSigning@1
+    inputs:
+      ConnectedServiceName: 'MSAL ESRP CodeSign Service Connection'
+      FolderPath: '$(Build.ArtifactStagingDirectory)'
+      Pattern: 'MSAL.zip'
+      signConfigType: 'inlineSignParams'
+      SessionTimeout: '60'
+      MaxConcurrency: '50'
+      MaxRetryAttempts: '5'
+      UseMinimatch: false
+      inlineOperation: |
+        [
+          {
+            "keyCode": "CP-233039-Apple",
```

Comment:
Is this a secret?

Reply:
Not a secret, it is just a code to specify what type of code signing needs to be performed by ESRP